helium/settings: add autocomplete options for search suggestions

potatosalad775 opened the pull request:
For your pull request to not get closed without review, please confirm that:

- [x] An issue exists where the maintainers agreed that this should be implemented.
      If such issue did not exist before, I opened one.
- [x] I tested that my contribution works locally, and does not break anything,
      otherwise I have marked my PR as draft.
- [x] If my contribution is non-trivial, I did not use AI to write most of it.
- [x] I understand that I will be permanently banned from interacting with this
      organization if I lied by checking any of these checkboxes.

Tested on (check one or more):
- [ ] Windows
- [x] macOS
- [ ] Linux

---

This PR adds autocomplete options for omnibox search suggestions.
Partially resolves #89 and #229 (duplicate)

<img width="1146" height="523" alt="img" src="https://github.com/user-attachments/assets/c1b19cf9-24b8-4535-b001-acb4946b4647" />

Tested on M4 Mac Mini, Tahoe 26.1, Helium 0.6.5.1

dumbmoron reviewed (dismissed):
lgtm, thanks

wukko reviewed (changes requested):
see my previous suggestions, we should resolve those before merging

potatosalad775 committed:
helium/core: add local omnibox suggestion options (#471)
- add option to exclude bookmark / browsing history from omnibox suggestions
- merge 'update-search-suggest-text.patch' into the new 'omnibox-suggestions.patch'

potatosalad775 commented on patches/helium/core/omnibox-suggestions.patch:
```diff
@@ -0,0 +1,313 @@
++  // responsible for navigating to exact urls (i.e. https://example.com/).
++  // If the user disabled history autocomplete, we need to disable just that,
++  // not the entire provider.
++  search_url_database_ = search_url_database_ &&
++                         client()->GetPrefs()->GetBoolean(
++                             omnibox::kAutocompleteHistoryEnabled);
```
This is probably a false positive.
Toggling the history suggestion feature on and off works fine on the actual development build.
